### ISSUE TITLE
The ServiceAccount field in Import Tekton resources from repository is no longer optional

### ISSUE THREAD

**Issue by rnapoles-tktn** [tektoncd/dashboard#1319]
**Problem:
The ServiceAccount field in Import Tekton resources from repository is identified as optional:
![image.png](https://images.zenhubusercontent.com/5db346d5b0111c0001df287a/68608b98-4134-4762-b33b-5cbb7b4e289f)

If you do NOT select a Service Account, then the corresponding pipelineRun will fail with:
`error: the path "/workspace/git-source/<APPLY-DIRECTORY>" does not exist`

**Versions:**
OCP 4.3, and 4.4-RC9
Dashboard/webhooks v0.5.3
Pipeline v0.10.1

**Expected:**
Should we be able to leave ServiceAccount empty?
If this is now mandatory then remove"(Optional)" from "ServiceAccount (Optional)" in the UI
Validate that a service account was selected
Should we be able to select multiple service accounts?

**Comment by a-roberts** [comment]
Yeah this is an interesting one, my guess is when no SA is specified it'll try default which won't have permissions, or pipeline if installed through the operator

I propose removing the optional part anyway because why would you want to be importing resources without knowledge of the service account that'll be used. For the short term I've added a note in the 0.6.1 release about this.

**Comment by a-roberts** [comment]
https://github.com/tektoncd/dashboard/pull/1320#issuecomment-619933247 latest discussion here, think we should hold off on this one until there's consensus, @eddycharly makes a good point. Added to the WG agenda for today.

**Comment by rhd-gitops-functional** [comment]
Had a ponder over this

```
error: the path "/workspace/git-source/<APPLY-DIRECTORY>" does not exist
```

so that's likely due to permission problems on the default service account that gets used on OpenShift, but in theory a user could configure this in such a way for it to work I'm sure. So the optionalness really is a thing (and you don't need to provide one on Docker Desktop).

I'm wondering if we instead do nothing on this, or perhaps add a tooltip - something to make it more obvious what's going on.

@AlanGreene @eddycharly your thoughts? It's on the agenda today but happy to discuss here/update if you folks can't make it.

**Comment by a-roberts** [comment]
Had a ponder over this

```
error: the path "/workspace/git-source/<APPLY-DIRECTORY>" does not exist
```

so that's likely due to permission problems on the default service account that gets used on OpenShift, but in theory a user could configure this in such a way for it to work I'm sure. So the optionalness really is a thing (and you don't need to provide one on Docker Desktop).

I'm wondering if we instead do nothing on this, or perhaps add a tooltip - something to make it more obvious what's going on.

@AlanGreene @eddycharly your thoughts? It's on the agenda today but happy to discuss here/update if you folks can't make it.

(also, if you're confused by the emails you just got - yeah, I accidentally commented on a functional account, mb)

**Comment by a-roberts** [comment]
Discussed on the WG call, @AlanGreene mentioned there is a ConfigMap that lets you specify the default ServiceAccount so making it mandatory doesn't make sense. Instead we can go with a Tooltip that calls out this feature and we can mention to users that it'll need sufficient permissions to create PipelineRuns and PipelineResources.

**Comment by AlanGreene** [comment]
default-service-account: https://github.com/tektoncd/pipeline/blob/eb26f85613d8fc73091ac5974bf0f44c0883592f/config/config-defaults.yaml#L43

also referenced in the install doc:
https://github.com/tektoncd/pipeline/blob/d882a824febdf766b45dbf99278ad0fb27432d65/docs/install.md#customizing-basic-execution-parameters

**Comment by a-roberts** [comment]
Simpler approach at https://github.com/tektoncd/dashboard/pull/1326
